lightwood/encoders/time_series/rnn.py

```diff
@@ -0,0 +1,178 @@
+            
+        self._prepared = True
+
+    def encode_one(self, data, initial_hidden = None, as_list = False, return_next_value = False):
+        """
+        This method encodes one single row of serial data
+        :param data:  a list of values [1,2,3,4,...]
+        :param initial_hidden: if you want to encode from an initial hidden state other than 0s
+        :param as_list: if you want to return the information as lists
+        :param return_next_value:  if you want to return the next value in the time series too
+
+        :return:  either encoded_value or encoded_value, next_value
+        """
+        if not self._prepared:
+            raise Exception('You need to call "prepare_encoder" before calling "encode" or "decode".')
+
+        self._encoder.eval()
+        with torch.no_grad():
+
+            data_tensor = tensor_from_series(data)
+
+            encoder_hidden = self._encoder.initHidden()
+            encoder_hidden = encoder_hidden if initial_hidden is None else initial_hidden
+            if type(encoder_hidden) is list:
+                encoder_hidden = torch.Tensor([[encoder_hidden]], device=device)
+            next_tensor = None
+            for tensor_i in range(len(data)):
+
+                next_tensor, encoder_hidden = self._encoder.forward(data_tensor[tensor_i], encoder_hidden)
+
+        if as_list:
+            if return_next_value:
+                return encoder_hidden.squeeze().tolist(), next_tensor.squeeze().tolist()
+            else:
+                return encoder_hidden.squeeze().tolist()
+
+        if return_next_value:
+            return encoder_hidden, next_tensor
+        else:
+            return encoder_hidden
+
+
+    def encode(self, column_data, get_next_count = None):
+        """
+        Encode a list of time series data
+        :param column_data: a list of lists [ [ts_data_point_list1], ...] to encode
+        :param get_next_count: default None, but you can pass a number X and it will return the X following predictions on the series for each ts_data_point in column_data
+        :return: a list of encoded values or if get_next_count !=0 two lists encoded_values, projected_numbers
+        """
+
+        if not self._prepared:
+            raise Exception('You need to call "prepare_encoder" before calling "encode" or "decode".')
+
+
+        ret = []
+        next = []
+
+        for val in column_data:
+            if get_next_count is None:
+                encoded = self.encode_one(val)
+            else:
+                if get_next_count <= 0:
+                    raise Exception('get_next_count must be greater than 0')
+
+                hidden = None
+                vector = val
+
+                next_i = []
+
+                for j in range(get_next_count):
+
+                    hidden, next_reading = self.encode_one(vector, initial_hidden=hidden, as_list=True, return_next_value=True)
```

Comment:
I like the list, it lets you test, and its optional, it will come handy to test when we build the decoder

Reply:
Fair enough, let's leave the argument but not use the list inside the encode function itself in that case (to speed things up).

Reply:
the thing is that RNNs are interesting in the sense that  since the usually run on a row by row basis unless we knew that all the dequencences were the same length we cannot do batches, still you are going to have to do append calls, not sure if tolist() is any more expensive than the append to tensor operations, but feel free to give it a crack if you rather do those operations on the tensor space.